core/src/main/java/org/owasp/dependencycheck/analyzer/PipfileAnalyzer.java

```diff
@@ -0,0 +1,201 @@
+     * enabled property.
+     *
+     * @return the analyzer's enabled property setting key
+     */
+    @Override
+    protected String getAnalyzerEnabledSettingKey() {
+        return Settings.KEYS.ANALYZER_PIPFILE_ENABLED;
+    }
+
+    @Override
+    protected void analyzeDependency(Dependency dependency, Engine engine) throws AnalysisException {
+        LOGGER.debug("Checking file {}", dependency.getActualFilePath());
+
+        if (REQUIREMENTS.equals(dependency.getFileName())) {
+            engine.removeDependency(dependency);
+        }
+        final File dependencyFile = dependency.getActualFile();
+        if (!dependencyFile.isFile() || dependencyFile.length() == 0) {
+            return;
+        }
+
+        final File actualFile = dependency.getActualFile();
+        if (actualFile.getName().equals(REQUIREMENTS)) {
+            final String contents = getFileContents(actualFile);
+            if (!contents.isEmpty()) {
+                final Matcher matcher = PACKAGE_VERSION.matcher(contents);
+                while (matcher.find()) {
+                    final String identifiedPackage = matcher.group(1);
+                    final String identifiedVersion = matcher.group(2);
+                    LOGGER.debug(String.format("package, version: %s %s", identifiedPackage, identifiedVersion));
+                    final Dependency d = new Dependency(dependency.getActualFile(), true);
+                    d.setName(identifiedPackage);
+                    d.setVersion(identifiedVersion);
+                    try {
+                        final PackageURL purl = PackageURLBuilder.aPackageURL()
+                                .withType("pypi")
+                                .withName(identifiedPackage)
+                                .withVersion(identifiedVersion)
+                                .build();
+                        d.addSoftwareIdentifier(new PurlIdentifier(purl, Confidence.HIGHEST));
+                    } catch (MalformedPackageURLException ex) {
+                        LOGGER.debug("Unable to build package url for pypi", ex);
+                        d.addSoftwareIdentifier(new GenericIdentifier("pypi:" + identifiedPackage + "@" + identifiedVersion, Confidence.HIGH));
+                    }
+                    d.setPackagePath(String.format("%s:%s", identifiedPackage, identifiedVersion));
+                    d.setEcosystem(PythonDistributionAnalyzer.DEPENDENCY_ECOSYSTEM);
+                    final String filePath = String.format("%s:%s/%s", dependency.getFilePath(), identifiedPackage, identifiedVersion);
+                    d.setFilePath(filePath);
+                    d.setSha1sum(Checksum.getSHA1Checksum(filePath));
+                    d.setSha256sum(Checksum.getSHA256Checksum(filePath));
+                    d.setMd5sum(Checksum.getMD5Checksum(filePath));
+                    d.addEvidence(EvidenceType.PRODUCT, REQUIREMENTS, "product", identifiedPackage, Confidence.HIGHEST);
```

Comment:
Unfortunately - there is a requirement to have at least some vendor evidence. In a case like this where we don't have a vendor the recommendation is to just put the product name in as the vendor.

```suggestion
                    d.addEvidence(EvidenceType.VENDOR, REQUIREMENTS, "vendor", identifiedPackage, Confidence.HIGHEST);
                    d.addEvidence(EvidenceType.PRODUCT, REQUIREMENTS, "product", identifiedPackage, Confidence.HIGHEST);
```